spec/models/sup_spec.rb

```diff
@@ -103,5 +103,29 @@
       sup = team.sups.first
       expect(team.users).to include sup.captain
     end
+
+    context 'find best captain' do
```

Comment:
Don't see an obvious problem, but write separate specs for `User#last_captain_at`, then `Sup#get_best_captain` and finally the ones that are side effect of `sup!`. I suspect there's a bug I am not seeing actually.